WalletWasabi/Nostr/NostrCoordinatorPublisher.cs

```diff
@@ -0,0 +1,49 @@
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin.Secp256k1;
+using NNostr.Client;
+using WalletWasabi.Bases;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Nostr;
+
+public class NostrCoordinatorPublisher : PeriodicRunner
+{
+	private readonly Uri[] _relayUris = [new("wss://relay.primal.net")];
+
+	public NostrCoordinatorPublisher(TimeSpan period, ECPrivKey key, NostrCoordinator coordinator) : base(period)
+	{
+		Coordinator = coordinator;
+		Key = key;
+		Client = NostrExtensions.Create(_relayUris, (EndPoint?)null);
+	}
+
+	private INostrClient Client { get; }
+
+	private NostrCoordinator Coordinator { get; }
+
+	private ECPrivKey Key { get; }
+
+	protected override async Task ActionAsync(CancellationToken cancel)
+	{
+		try
+		{
+			var discoveryEvent = await Key.CreateCoordinatorDiscoveryEventAsync(Coordinator).ConfigureAwait(false);
+
+			using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
+			using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cts.Token, cancel);
+			await Client.PublishAsync([discoveryEvent], linkedCts.Token).ConfigureAwait(false);
+		}
+		catch (Exception ex) when (ex is not OperationCanceledException)
+		{
+			Logger.LogError(ex);
+		}
```

Comment:
No. The only reason we still use the `PeriodicRunner` instead of something better is because it handles exceptions automatically. Remove the `try-catch` block.